gcloud/pubsub/_gax.py

```diff
@@ -0,0 +1,190 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""GAX wrapper for Pubsub API requests."""
+try:
+    # pylint: disable=no-name-in-module
+    from google.gax import CallOptions
+    from google.gax.errors import GaxError
+    from google.gax.grpc import exc_to_code
+    # pylint: enable=no-name-in-module
+except ImportError:  # pragma: NO COVER
+    _HAVE_GAX = False
+else:
+    _HAVE_GAX = True
+
+    from google.pubsub.v1.pubsub_pb2 import PubsubMessage
+    from grpc.beta.interfaces import StatusCode
+
+    from gcloud.exceptions import Conflict
+    from gcloud.exceptions import NotFound
+    from gcloud._helpers import _to_bytes
+
+    class _PublisherAPI(object):
+        """Helper mapping publisher-related APIs.
+
+        :type gax_api: :class:`google.pubsub.v1.publisher_api.PublisherApi`
+        :param gax_api: API object used to make GAX requests.
+        """
+        def __init__(self, gax_api):
+            self._gax_api = gax_api
+
+        def list_topics(self, project):
+            """List topics for the project associated with this API.
+
+            See:
+            https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/list
+
+            :type project: string
+            :param project: project ID
+
+            :rtype: tuple, (list, str)
+            :returns: list of ``Topic`` resource dicts, plus a
+                    "next page token" string:  if not None, indicates that
+                    more topics can be retrieved with another call (pass that
+                    value as ``page_token``).
+            """
+            options = CallOptions(is_page_streaming=False)
+            path = 'projects/%s' % (project,)
+            response = self._gax_api.list_topics(path, options)
+            topics = [{'name': topic_pb.name} for topic_pb in response.topics]
+            return topics, response.next_page_token
+
+        def topic_create(self, topic_path):
+            """API call:  create a topic
+
+            See:
+            https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/create
+
+            :type topic_path: string
+            :param topic_path: fully-qualified path of the new topic, in format
+                               ``projects/<PROJECT>/topics/<TOPIC_NAME>``.
+
+            :rtype: dict
+            :returns: ``Topic`` resource returned from the API.
+            :raises: :exc:`gcloud.exceptions.Conflict` if the topic already
+                     exists
+            """
+            try:
+                topic_pb = self._gax_api.create_topic(topic_path)
+            except GaxError as exc:
+                if exc_to_code(exc.cause) == StatusCode.FAILED_PRECONDITION:
+                    raise Conflict(topic_path)
+                raise  # pragma: NO COVER
```

Comment:
Can you just add a unit test for this? We should use no-cover's as a last resort.